Set up Playwright screenshot testing

EthanThatOneKid opened the pull request:
### What we can do now

`npm t` has now been upgraded to include parallelized Playwright tests and screenshot diff testing. To specifically test the Playwright tests and not the Vitest tests, you may run `npm run test:browser`. Make some changes on screenshot-tested pages and give it a try!

```bash
npm t
```

### Change log
- Split up `/tests/test.ts` tests into each page of the `/src/routes/` directory.
- Added `projectsFromMatrix` helper to generate a matrix of screenshots to be taken per page.
- Added `vitest.config.ts` to exclude Playwright tests from the Vitest testing.
- Generated the first official set of screenshots in `/src/routes/screenshots/`.
- This PR goes on to add the screenshot testing boilerplate for each page endpoint of the website.
  - [x] Added screenshot tests for 404 error page.
  - [x] Added screenshot tests for `/1st` page.
  - [x] Added screenshot tests for `/about` page.
  - [x] Added screenshot tests for `/blog` page.
  - [x] Added screenshot tests for `/blog/:id` page.
  - [x] Added screenshot tests for `/events` page.
  - [x] Added screenshot tests for `/nodebuds` page.
  - [x] Added screenshot tests for `/privacy` page.
  - [x] Added screenshot tests for `/quiz` page.
  - [x] Added screenshot tests for `/teams` page.

Resolves #557.

EthanThatOneKid committed:
Progress

EthanThatOneKid committed:
Split up tests into multiple test files in `/src/routes/`

EthanThatOneKid committed:
Enabled 'parallel' mode for all Playwright test files

And added testing for `/blog/272`.

EthanThatOneKid committed:
Added screenshot testing to the about page

Also, slightly refactored `src\routes\(site)\about\officer-profile.svelte` to remove unused code.

EthanThatOneKid committed:
Added screenshot testing to `/1st`

And removed threshold of 0.1 since the default of 0.2 should be better. If not, we should bump it up to 0.3.

EthanThatOneKid committed:
Added screenshot testing to the `/blog` page

EthanThatOneKid committed:
Added screenshot tests for `/events` page

EthanThatOneKid committed:
Attempt to fix navbar inconsistency

Added waitFor: 'network-idle' in an attempt to wait for the JS to load.

Also removed tests for blog page since there is currently no way of running the Playwright tests with DEBUG_MODE_ENABLED sadly.

EthanThatOneKid committed:
Gave a larger threshold to Playwright snapshot tests

EthanThatOneKid committed:
Disabled CSS for image rendering

Replaced threshold with a starting `maxDiffPixelRatio: 0.1` and a lax threshold of `threshold: 1`.

EthanThatOneKid committed:
Added screenshot testing to `/teams`

EthanThatOneKid committed:
Added screenshot testing to `/quiz`

EthanThatOneKid committed:
Added screenshot testing for `/privacy`

EthanThatOneKid committed:
Added screenshot testing to `/nodebuds`

charliettaylor commented on package.json:
```diff
@@ -19,15 +19,15 @@
   },
   "devDependencies": {
-    "@playwright/test": "^1.25.0",
+    "@playwright/test": "^1.28.0-alpha-oct-12-2022",
```
Is there a specific reason for using this version? Using alpha seems unnecessary

EthanThatOneKid replied:
> Is there a specific reason for using this version? Using alpha seems unnecessary

Good catch. While testing, I updated to the `@next` version, but you are right that’s probably not necessary. I’ll set it to the `@latest` version instead.

EthanThatOneKid replied:
Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/commit/829a50bfa2b0cdad02a5d70ecfb0814d7e20da26>.

EthanThatOneKid committed:
Added jsenabled and jsdisabled to Playwright project matrix

This kinda took way too long to test :/
BUT screenshot tests are cool and convenient.

EthanThatOneKid committed:
Ran `npm i -D @playwright/test@latest`

Resolves <https://github.com/EthanThatOneKid/acmcsuf.com/pull/612#discussion_r993985151>.

EthanThatOneKid committed:
Decided on viewports for Playwright project matrix

- Mobile matches Samsung Galaxy S7 Edge
- Tablet matches Samsung Galaxy Tab 10

EthanThatOneKid committed:
Removed Build step from `check_website.yaml` workflow

k4rni reviewed (approved):
LGTM

jaasonw reviewed (approved):
i did not look at it so therefore i see no issues with it

charliettaylor reviewed (approved):
![image](https://cdn140.picsart.com/354090890050211.png)

EthanThatOneKid committed:
Merge branch 'main' into fix/557
